cvmfs/catalog_mgr_rw.cc

```diff
@@ -837,6 +847,34 @@ shash::Any WritableCatalogManager::SnapshotCatalog(WritableCatalog *catalog)
   return hash_catalog;
 }
 
+void WritableCatalogManager::Balance() {
+  CatalogList catalog_list = GetCatalogs();
+  for (unsigned i = 0; i < catalog_list.size(); ++i) {
+    FixWeight(static_cast<WritableCatalog*>(catalog_list[i]));
+  }
+}
+
+void WritableCatalogManager::FixWeight(WritableCatalog* catalog) {
+  // firstly check underflow because they can provoke overflows
+  if (catalog->GetNumEntries() < min_weight_ &&
+      !catalog->IsRoot() &&
+      catalog->IsAutogenerated()) {
+    LogCvmfs(kLogCatalog, kLogStdout,
+             "deleting an autogenerated catalog in '%s'",
+             catalog->path().c_str());
+    // Remove the .cvmfscatalog and .cvmfsautocatalog files first
+    string path = catalog->path().ToString();
+    catalog->RemoveEntry(path + "/.cvmfscatalog");
```

Comment:
Off topic, while I was coding this I thought it would be a good idea to have a class/namespace in a sole file that can be exported and would contain all this constants. It's not precisely a good practise to hardcode file names, because one day it can be renamed, and that day we would have to look in all the files to check wether they contain the ".cvmfscatalog" string or not. What do you think?